src/unstract/sdk/adapters/x2text/llm_whisperer/src/llm_whisperer.py

```diff
@@ -303,25 +313,49 @@ def _send_whisper_request(
     def _extract_text_from_response(
         self, output_file_path: Optional[str], response: requests.Response
     ) -> str:
-
-        output = ""
+        output_json = {}
         if response.status_code == 200:
-            output = response.content.decode("utf-8")
+            output_json = response.json()
         elif response.status_code == 202:
-            whisper_hash = response.json().get(WhisperStatus.WHISPER_HASH)
-            output = self._extract_async(whisper_hash=whisper_hash)
+            whisper_hash = response.json().get("WhisperStatus.WHISPER_HASH")
+            output_json = self._extract_async(whisper_hash=whisper_hash)
         else:
             raise ExtractorError("Couldn't extract text from file")
+        if output_file_path:
+            self._write_output_to_file(
+                output_json=output_json,
+                output_file_path=Path(output_file_path),
+            )
+        return output_json.get("text", "")
 
+    def _write_output_to_file(self, output_json: dict, output_file_path: Path):
         try:
-            # Write output to a file
-            if output_file_path:
-                with open(output_file_path, "w", encoding="utf-8") as f:
-                    f.write(output)
-        except OSError as e:
-            logger.error(f"OS error while writing {output_file_path}: {e} ")
+            text_output = output_json.get("text", "")
+            logger.info(f"Writing output to {output_file_path}")
+            output_file_path.write_text(text_output, encoding="utf-8")
+            try:
+                # Define the directory of the output file and metadata paths
+                output_dir = output_file_path.parent
+                metadata_dir = output_dir / "metadata"
+                metadata_file_name = output_file_path.with_suffix(".json").name
+                metadata_file_path = metadata_dir / metadata_file_name
+                # Ensure the metadata directory exists
+                metadata_dir.mkdir(parents=True, exist_ok=True)
+                # Remove the "text" key from the metadata
+                metadata = {
+                    key: value for key, value in output_json.items() if key != "text"
+                }
+                metadata_json = json.dumps(metadata, ensure_ascii=False, indent=4)
+                logger.info(f"Writing metadata to {metadata_file_path}")
+                metadata_file_path.write_text(metadata_json, encoding="utf-8")
+            except Exception as e:
+                logger.error(
+                    f"Error while writing metadata to {metadata_file_path}: {e}"
+                )
```

Comment:
Is the error suppression here intentional? Do we check elsewhere if metadata directory is present or not before attempting to access it?

Reply:
@chandrasekharan-zipstack We are creating metadata directory if not exists in line 343. Error suppression is intentional here because even if we don't have metadata for the file, extraction should still continue.